src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -8902,7 +8902,7 @@ bool Unit::IsInDisallowedMountForm() const
     CreatureModelDataEntry const* model = sCreatureModelDataStore.LookupEntry(display->ModelId);
     ChrRacesEntry const* race = sChrRacesStore.LookupEntry(displayExtra->Race);
 
-    if (model && !(model->Flags & 0x80))
+    if (model && !(model->HasFlag(CreatureModelDataFlags::CREATURE_MODEL_DATA_FLAGS_CAN_MOUNT)))
```

Comment:
Codestyle remark: dont prefix regular enums with `Type::`, use that only for `enum class` where its required